yt_dlp/extractor/twitter.py

```diff
@@ -503,90 +607,114 @@ def add_thumbnail(name, size):
-                        'ie_key': PeriscopeIE.ie_key(),
-                    })
-                elif card_name == 'broadcast':
-                    info.update({
-                        '_type': 'url',
-                        'url': get_binding_value('broadcast_url'),
-                        'ie_key': TwitterBroadcastIE.ie_key(),
-                    })
-                elif card_name == 'summary':
-                    info.update({
-                        '_type': 'url',
-                        'url': get_binding_value('card_url'),
-                    })
-                elif card_name == 'unified_card':
-                    media_entities = self._parse_json(get_binding_value('unified_card'), twid)['media_entities']
-                    extract_from_video_info(next(iter(media_entities.values())))
-                # amplify, promo_video_website, promo_video_convo, appplayer,
-                # video_direct_message, poll2choice_video, poll3choice_video,
-                # poll4choice_video, ...
-                else:
-                    is_amplify = card_name == 'amplify'
-                    vmap_url = get_binding_value('amplify_url_vmap') if is_amplify else get_binding_value('player_stream_url')
-                    content_id = get_binding_value('%s_content_id' % (card_name if is_amplify else 'player'))
-                    formats, subtitles = self._extract_formats_from_vmap_url(vmap_url, content_id or twid)
-                    self._sort_formats(formats)
-
-                    thumbnails = []
-                    for suffix in ('_small', '', '_large', '_x_large', '_original'):
-                        image = get_binding_value('player_image' + suffix) or {}
-                        image_url = image.get('url')
-                        if not image_url or '/player-placeholder' in image_url:
-                            continue
-                        thumbnails.append({
-                            'id': suffix[1:] if suffix else 'medium',
-                            'url': image_url,
-                            'width': int_or_none(image.get('width')),
-                            'height': int_or_none(image.get('height')),
-                        })
-
-                    info.update({
-                        'formats': formats,
-                        'subtitles': subtitles,
-                        'thumbnails': thumbnails,
-                        'duration': int_or_none(get_binding_value(
-                            'content_duration_seconds')),
-                    })
-            else:
-                expanded_url = try_get(status, lambda x: x['entities']['urls'][0]['expanded_url'])
-                if not expanded_url:
-                    raise ExtractorError("There's no video in this tweet.")
-                info.update({
+        def extract_from_card_info(card):
+            self.write_debug(f'Extracting from card info: {card.get("url")}')
```

Comment:
unnecessary imo

Reply:
I thought it might give a better trace for people debugging problems as its analogue to the video extraction information in `extract_from_video_info`.